zookeeper-server/src/main/java/org/apache/zookeeper/server/NettyServerCnxnFactory.java

```diff
@@ -423,15 +425,29 @@ public void operationComplete(Future<Channel> future) {
 
                     String authProviderProp = System.getProperty(x509Util.getSslAuthProviderProperty(), "x509");
 
-                    X509AuthenticationProvider authProvider = (X509AuthenticationProvider) ProviderRegistry.getProvider(authProviderProp);
+                    // All customer variatoins of AuthenticationProvider should be supported here. Currently
```

Comment:
Fix typo - also a better way to say this here is "All implementations of the AuthenticationProvider interface should be supported here"